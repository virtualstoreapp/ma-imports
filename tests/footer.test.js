const fs = require('fs');
const path = require('path');
require('@testing-library/jest-dom');

describe('Footer CSS and Markup', () => {
  it('includes a CSS rule to center footer text', () => {
    const cssPath = path.resolve(__dirname, '../styles.css');
    const cssContent = fs.readFileSync(cssPath, 'utf8');
    const normalized = cssContent.replace(/\s/g, '');
    expect(normalized).toMatch(/footer\{[^}]*text-align:center;[^}]*\}/);
  });

  it('renders footer markup as expected', () => {
    const htmlPath = path.resolve(__dirname, '../index.html');
    const htmlContent = fs.readFileSync(htmlPath, 'utf8');
    document.body.innerHTML = htmlContent;
    const footer = document.querySelector('footer');
    expect(footer).toBeInTheDocument();
    expect(footer.outerHTML).toMatchSnapshot();
  });
});
