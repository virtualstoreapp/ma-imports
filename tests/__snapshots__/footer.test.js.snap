// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer CSS and Markup renders footer markup as expected 1`] = `
"<footer>
    <p>Vers√£o: 2.4.0 - 0.0.97</p>
  </footer>"
`;
